src/util/worker/mod.rs

```diff
@@ -254,14 +273,10 @@ fn poll<R, T, U>(
 
         keep_going = fill_task_batch(&rx, &mut batch, batch_size, timeout);
         if !batch.is_empty() {
-            counter.fetch_sub(batch.len(), Ordering::SeqCst);
-            WORKER_PENDING_TASK_VEC
-                .with_label_values(&[&name])
-                .sub(batch.len() as i64);
-            WORKER_HANDLED_TASK_VEC
-                .with_label_values(&[&name])
-                .inc_by(batch.len() as i64);
             runner.run_batch(&mut batch);
+            counter.fetch_sub(batch.len(), Ordering::SeqCst);
```

Comment:
note: we change the order (run first, then decrease the pending_count) in order to keep the same semantic as  future worker and future pool.